packages/sanity/src/core/form/inputs/files/ImageToolInput/imagetool/draw.ts

```diff
@@ -0,0 +1,379 @@
+
+  context.restore()
+
+  function vline(x: number) {
+    line(x, margin, x, image.height - margin)
+  }
+
+  function hline(y: number) {
+    line(margin, y, image.width - margin, y)
+  }
+
+  function line(x1: number, y1: number, x2: number, y2: number) {
+    context.beginPath()
+    context.moveTo(x1, y1)
+    context.lineTo(x2, y2)
+    context.stroke()
+    context.closePath()
+  }
+}
+
+function paintCropBorder({
+  context,
+  cropRect,
+}: {
+  context: CanvasRenderingContext2D
+  cropRect: Rect
+}): void {
+  context.save()
+  context.beginPath()
+  context.fillStyle = 'rgba(66, 66, 66, 0.9)'
+  context.lineWidth = 1
+  context.rect(cropRect.left, cropRect.top, cropRect.width, cropRect.height)
+  context.stroke()
+  context.closePath()
+  context.restore()
+}
+
+function highlightCropHandles({
+  context,
+  cropHandles,
+  cropping,
+  opacity,
+}: {
+  context: CanvasRenderingContext2D
+  cropHandles: CropHandles
+  cropping: keyof CropHandles | false
+  opacity: number
+}): void {
+  context.save()
+
+  //context.globalCompositeOperation = "difference";
+
+  cropHandleKeys.forEach((handle) => {
+    context.fillStyle =
+      cropping === handle
+        ? `rgba(202, 54, 53, ${opacity})`
+        : `rgba(230, 230, 230, ${opacity + 0.4})`
+    const {left, top, height, width} = cropHandles[handle]
+    context.fillRect(left, top, width, height)
+    context.beginPath()
+    context.fillStyle = `rgba(66, 66, 66, ${opacity})`
+    context.rect(left, top, width, height)
+    context.closePath()
+    context.stroke()
+  })
+  context.restore()
+}
+
+/** @internal */
+export const cropHandleKeys: (keyof CropHandles)[] = [
+  'left',
+  'right',
+  'top',
+  'topLeft',
+  'topRight',
+  'bottom',
+  'bottomLeft',
+  'bottomRight',
+]
+
+/** @internal */
+export function paint({
+  clampedValue,
+  context,
+  cropHandles,
+  cropping,
+  cropRect,
+  HOTSPOT_HANDLE_SIZE,
+  hotspotRect,
+  image,
+  MARGIN_PX,
+  pointerPosition,
+  ratio,
+  readOnly,
+  scale,
+}: {
+  clampedValue: {crop: Rect; hotspot: Rect}
+  context: CanvasRenderingContext2D
+  cropHandles: CropHandles
+  cropping: keyof CropHandles | false
+  cropRect: Rect
+  HOTSPOT_HANDLE_SIZE: number
+  hotspotRect: Rect
+  image: HTMLCanvasElement
+  MARGIN_PX: number
+  pointerPosition: Coordinate | null
+  ratio: number
+  readOnly: boolean
+  scale: number
+}): void {
+  context.save()
+
+  context.scale(ratio, ratio)
+
+  const opacity = !readOnly && pointerPosition ? 0.8 : 0.2
+
+  paintBackground({context, image, MARGIN_PX, scale})
+  paintHotspot({
+    clampedValue,
+    context,
+    HOTSPOT_HANDLE_SIZE,
+    image,
+    MARGIN_PX,
+    opacity,
+    readOnly,
+    scale,
+  })
+  printGuidelines({context, hotspotRect, image, MARGIN_PX, scale})
+  paintCropBorder({context, cropRect})
+
+  if (!readOnly) {
+    highlightCropHandles({context, cropHandles, cropping, opacity})
+  }
+
+  // paintPointerPosition({context, pointerPosition, scale})
```

Comment:
I believe this is old code but this doesn't look intended

Reply:
Yeah I wasn't sure if it should be removed so I ported it with the rest 🤷